src/ut/resolver_test.cpp

```diff
@@ -0,0 +1,127 @@
+ * option) any later version, along with the "Special Exception" for use of
+ * the program along with SSL, set forth below. This program is distributed
+ * in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR
+ * A PARTICULAR PURPOSE.  See the GNU General Public License for more
+ * details. You should have received a copy of the GNU General Public
+ * License along with this program.  If not, see
+ * <http://www.gnu.org/licenses/>.
+ *
+ * The author can be reached by email at clearwater@metaswitch.com or by
+ * post at Metaswitch Networks Ltd, 100 Church St, Enfield EN2 6BQ, UK
+ *
+ * Special Exception
+ * Metaswitch Networks Ltd  grants you permission to copy, modify,
+ * propagate, and distribute a work formed by combining OpenSSL with The
+ * Software, or a work derivative of such a combination, even if such
+ * copying, modification, propagation, or distribution would otherwise
+ * violate the terms of the GPL. You must comply with the GPL in all
+ * respects for all of the code used other than OpenSSL.
+ * "OpenSSL" means OpenSSL toolkit software distributed by the OpenSSL
+ * Project and licensed under the OpenSSL Licenses, or a work based on such
+ * software and licensed under the OpenSSL Licenses.
+ * "OpenSSL Licenses" means the OpenSSL License and Original SSLeay License
+ * under which the OpenSSL Project distributes the OpenSSL toolkit software,
+ * as those licenses appear in the file LICENSE-OPENSSL.
+ */
+
+#include<resolver_test.h>
+#include "resolver_utils.h"
+#include "test_interposer.hpp"
+
+/// DNS Resolver is created with server address 0.0.0.0 to disable server
+/// queries.
+ResolverTest::ResolverTest() :
+  _dnsresolver("0.0.0.0")
+{
+  cwtest_completely_control_time();
+}
+
+ResolverTest::~ResolverTest()
+{
+  cwtest_reset_time();
+}
+
+AddrInfo ResolverTest::ip_to_addr_info(std::string address_str, int port, int transport)
+{
+  AddrInfo ai;
+  BaseResolver::parse_ip_target(address_str, ai.address);
+  ai.port = TEST_PORT;
+  ai.transport = TEST_TRANSPORT;
+  return ai;
+}
+
+void ResolverTest::add_white_records(int count, std::string host)
+{
+  std::vector<DnsRRecord*> records;
+  std::stringstream os;
+  for (int i = 0; i < count; ++i)
+  {
+    os << "3.0.0." << i;
+    records.push_back(ResolverUtils::a(host, 3600, os.str()));
+    os.str(std::string());
+  }
+  _dnsresolver.add_to_cache(host, ns_t_a, records);
+}
+
+bool ResolverTest::resolution_contains(AddrInfo ai, int max_targets)
+{
+  std::vector<AddrInfo> targets = resolve(max_targets);
+  std::vector<AddrInfo>::const_iterator record_iterator;
+  record_iterator = find(targets.begin(), targets.end(), ai);
+  return (record_iterator != targets.end());
+}
+
+bool ResolverTest::is_black(std::string address_str, int count, int repetitions)
+{
+  AddrInfo ai = ip_to_addr_info(address_str);
+  // The black record should not be returned on any call.
```

Comment:
I think it would be useful to explain your strategy a *little* bit more (i.e. explaining why you select count-1 records, and why you repeat).